tests/tracking/test_mlflow_artifacts.py

```diff
@@ -0,0 +1,210 @@
+import os
+from collections import namedtuple
+import subprocess
+import tempfile
+import requests
+
+import pytest
+
+import mlflow
+from tests.helper_functions import LOCALHOST, get_safe_port
+from tests.tracking.integration_test_utils import _await_server_up_or_die
+
+
+def _launch_server(backend_store_uri, artifacts_destination):
+    port = get_safe_port()
+    url = f"http://{LOCALHOST}:{port}"
+    extra_cmd = ["--gunicorn-opts", "--log-level debug"] if os.name == "posix" else []
+    cmd = [
+        "mlflow",
+        "server",
+        "--backend-store-uri",
+        backend_store_uri,
+        "--default-artifact-root",
+        f"{url}/api/2.0/mlflow-artifacts/artifacts",
+        "--artifacts-destination",
+        artifacts_destination,
+        "--host",
+        LOCALHOST,
+        "--port",
+        str(port),
+        *extra_cmd,
+    ]
+    process = subprocess.Popen(cmd)
+    _await_server_up_or_die(port)
+    return url, process
+
+
+ArtifactsServer = namedtuple(
+    "ArtifactsServer", ["backend_store_uri", "artifacts_destination", "url", "process"]
+)
+
+
+@pytest.fixture(scope="module")
+def artifacts_server():
+    with tempfile.TemporaryDirectory() as tmpdir:
+        backend_store_uri = f"{tmpdir}/mlruns"
+        artifacts_destination = f"{tmpdir}/artifacts"
+        url, process = _launch_server(backend_store_uri, artifacts_destination)
+        yield ArtifactsServer(backend_store_uri, artifacts_destination, url, process)
+        process.kill()
+
+
+def read_file(path):
+    with open(path) as f:
+        return f.read()
+
+
+def upload_file(url, local_path):
+    with open(local_path, "rb") as f:
+        requests.put(url, data=f).raise_for_status()
+
+
+def download_file(url, local_path):
+    with requests.get(url, stream=True) as r:
+        r.raise_for_status()
+        with open(local_path, "wb") as f:
+            for chunk in r.iter_content(chunk_size=8192):
```

Comment:
Where are we getting the chunk size of 3KB from? Should we potentially choose a larger chunk size?